packages/cookie-consent/src/CookieConsentProvider/useSegmentIntegrations.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react'
+import type { CategoryKind, Config, Integrations } from './types'
+
+type SegmentIntegration = {
+  category: string
+  creationName: string
+  description: string
+  name: string
+  website: string
+}
+
+const defaultSegmentIoIntegration: SegmentIntegration = {
+  category: 'Functional',
+  creationName: 'Segment.io',
+  description: '',
+  name: 'Segment.io',
+  website: 'https://segment.io',
+}
+
+const timeout = (time: number) => {
+  const controller = new AbortController()
+  setTimeout(() => controller.abort(), time * 1000)
+
+  return controller
+}
+
+type SegmentIntegrations = SegmentIntegration[]
+
+const CATEGORY_MATCH: Record<string, CategoryKind> = {
+  Analytics: 'analytics',
+  CRM: 'marketing',
+  Other: 'marketing',
+  Functional: 'functional',
+}
+
+const transformSegmentIntegrationsToIntegrations = (
+  segmentIntegrations: SegmentIntegrations,
+): Integrations =>
+  [defaultSegmentIoIntegration, ...segmentIntegrations].map(
+    ({ name, category, creationName }) => ({
+      // Segment requires the `creationName` for this destination.
+      // This condition is a test (as of 2023-02-28)
+      // and should either be improved or deleted.
+      name: name === 'Google Ads (Gtag)' ? creationName : name,
+      category: CATEGORY_MATCH[category] ?? 'marketing',
+    }),
+  )
```

Comment:
We initially added this condition to verify the creationName was the proper slug to send back to Segment. IMO we should simplify and always send the creationName, which is the recommended way by Segment.

```suggestion
  [defaultSegmentIoIntegration, ...segmentIntegrations].map(
    ({ category, creationName }) => ({
      name: creationName,
      category: CATEGORY_MATCH[category] ?? 'marketing',
    }),
  )
```

Would need to be monitor, but I guess it would be a good timing if we include this package.